dataland-frontend/src/components/resources/frameworkDataSearch/lksg/LksgPanel.vue

```diff
@@ -180,12 +192,69 @@ export default defineComponent({
      */
     reformatCountriesValue(kpiValue: KpiValue) {
       return Array.isArray(kpiValue)
-        ? kpiValue.map(
-            (countryCodeShort: string) => getCountryNameFromCountryCode(countryCodeShort) ?? countryCodeShort
-          )
+        ? kpiValue.map((countryCodeShort: string) => getCountryNameFromCountryCode(countryCodeShort))
         : getCountryNameFromCountryCode(kpiValue as string) ?? kpiValue;
     },
 
+    /**
+     * Generates a list of readable strings (or just a single one) combining suppliers and their associated countries
+     * @param numberOfSuppliersPerCountryCode the map of number of suppliers and associated companies
+     * from which strings are written
+     * @returns the constructed collection of readable strings
+     */
+    generateReadableCombinationOfNumberOfSuppliersAndCountries(
+      numberOfSuppliersPerCountryCode?: Map<string, number | undefined | null>
+    ) {
+      if (numberOfSuppliersPerCountryCode != undefined) {
+        const readableListOfSuppliersAndCountries = Array.from(numberOfSuppliersPerCountryCode.entries()).map(
+          ([countryCode, numberOfSuppliers]) => {
+            const countryName = getCountryNameFromCountryCode(countryCode);
+            if (numberOfSuppliers != undefined) {
+              return String(numberOfSuppliers) + " suppliers from " + countryName;
+            } else {
+              return "There are suppliers from " + countryName;
+            }
+          }
+        );
+        if (readableListOfSuppliersAndCountries.length > 1) {
+          return readableListOfSuppliersAndCountries;
+        } else {
+          return readableListOfSuppliersAndCountries[0];
+        }
+      } else {
+        return null;
+      }
+    },
+
+    /**
+     * Converts the map of ProcurementCategory and LksgProductCategory into an array for a proper handling of the
+     * DetailsCompanyDataTable in the LksgCompanyDataTable (modal showing information related to Procurement Categories)
+     * @param inputObject Map to convert to array
+     * @returns The constructed map
+     */
+    reformatProcurementCategoriesValue(inputObject: Map<ProcurementCategoryType, LksgProcurementCategory> | null) {
+      if (inputObject == null) return null;
+      const inputObjectEntries = Object.entries(inputObject) as [string, LksgProcurementCategory][];
+      return inputObjectEntries.map(([procurementCategoryType, lksgProcurementCategory]) => {
+        const definitionsOfProductTypeOrService =
+          lksgProcurementCategory.procuredProductTypesAndServicesNaceCodes.length > 1
+            ? lksgProcurementCategory.procuredProductTypesAndServicesNaceCodes
+            : lksgProcurementCategory.procuredProductTypesAndServicesNaceCodes[0] || "";
```

Comment:
This is actually an excelent case to use the `??` operator:

`lksgProcurementCategory.procuredProductTypesAndServicesNaceCodes[0] ?? "";`

Reply:
Done.

Reply:
👍